### ISSUE TITLE
Certain files don't get updated on workshop

### ISSUE THREAD

**Issue by etet2** [Regalis11/Barotrauma#1534]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
For some reason whever you push a mod update onto the forum, it refuses to update certain files and instead later on the next workshop sync/update(or however else it works) downloads old ones onto your computer reverting all of the progress

**Steps To Reproduce**
1. Change files on your system
2. Go onto ingame workshop and press update item
3. Restart the game couple of times or manually press the update button on subscribed mods page
i'm not sure how it picks what files to carry over and what files to glitch out and not update
**Additional information**

**Comment by NilanthAnimosus** [comment]
Seen this issue in action, its definately a bug that exists.

I had a workshop item update and went to update it ingame, it actually removed files he made both in the steam workshop folder and my game folder.

Upon trying to revert/fix these issues, he lost two files, having to remake one while I thought (based on this issue) that problems may arise and made a copy, though my copy was already missing one.

Additionally my game updated a steam workshop item in game but even after restarting barotrauma it did not correctly update the contentpackage definition for it.

I also managed to get an error where the contentpackage xml is already there thus I couldn't enable it in all this mess as well - there are some workshop bugs present.

**Comment by NilanthAnimosus** [comment]
Could we not simply directly use the content (like most other games) from the workshop directory itself? or copy the workshop folder -exactly- and find the filelist.xml from the folder ported to ../Mods in a more sensible manner? I feel it would be less buggy and make more sense if it was more just a 1:1 copy without changing what goes where, and simply change anything that is different.

I feel there is a greatly simpler way of handling this.

**Comment by itchyOwl** [comment]
Not sure if directly related or a separate issue, but let's post it here:

"Well this is interesting. I was using content downloaded from the workshop, but I unsubscribed from them all yesterday. However all of the content package files for them are still here. Including one called Mallory . I'll try to launch the game with mallory .xml removed from the folder, see if it no longer gives me the error"

https://cdn.discordapp.com/attachments/587647684794515468/587684489237364880/mallory.png

"I forced steam to update my workshop downloads after removing mallory's xml and the game is no longer attempting to load any of them; which solves my original problem. The files for old subscriptions are still present in the folder but It sounds like you folks are already aware of that so I'm up to consider this "ticket" closed"

https://cdn.discordapp.com/attachments/586554844941123604/587989607984005121/unknown.png
